src/main/java/innohack/gem/entity/rule/Parameter.java

```diff
@@ -51,4 +51,31 @@ public ParamType getType() {
   public void setType(ParamType type) {
     this.type = type;
   }
+
+  @Override
+  public int compareTo(Parameter parameter) {
+    if (this.equals(parameter)) {
+      return 0;
+    } else {
+      return -1;
```

Comment:
This comparator looks odd and non-deterministic.. maybe need to either compare by the label or value?